content/doc/developer/internationalization/internationalization-and-localization.adoc

```diff
@@ -0,0 +1,320 @@
+the remainder of this section, we discuss what needs to be translated
+and how.
+
+[cols="",]
+|===
+|Remember that properties files must be encoded in
+https://docs.oracle.com/javase/6/docs/api/java/util/Properties.html[ISO
+8859-1]. This should be the default anyway.
+|===
+
+=== Translation tools
+
+* https://wiki.jenkins.io/display/JENKINS/Translation-Tool[Translation
+Tool] : Command line tool to help translators to generate files, add
+missing keys, and more.
+* https://plugins.jenkins.io/translation/[Translation
+Assistance Plugin] : Adds a dialog box in Jenkins to translate and send
+missing keys.
+
+=== Translating `+Messages.properties+`
+
+Developers place messages that require localization in
+`+Messages.properties+`. Those need to be translated in the usual
+manner. See `+Messages_ja.properties+` in the core as an example if you
+are new to this process.
+
+If looking at `+Messages.properties+` alone does not give you
+enough contextual information about where the messages are used. you can access messages by with the
+type-safe `+Messages+` class generated by
+https://docs.oracle.com/middleware/1212/wls/WLAPI/weblogic/i18n/Localizer.html[Localizer]. To find out where
+messages are actually used, use your IDE to find all the usages of the
+message format method.
+
+=== Translating message references in Jelly
+
+The other messages that need to be translated are in Jelly view files,
+which are in `+src/main/resources/**.jelly+`. To localize them, first
+you run Maven to generate a skeleton property file for your locale:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+$ cd jenkins/core  (or a plugin dir)
+$ mvn stapler:i18n -Dlocale=fr
+....
+
+|===
+
+This generates `+*_fr.properties+` throughout
+`+src/main/resources+` with an empty value. If the file already exists,
+it appends missing entries to existing files.
+
+You can then translate messages in each property file.
+It is not necessary to translate the entire file;
+entries that are empty fall back to the default locale.
+
+[cols="",]
+|===
+|https://addons.mozilla.org/en-US/firefox/addon/locale-switcher/[Quick Locale Switcher]
+extension to firefox is useful to toggle between various locales.
+|===
+
+=== Translating static HTML resources
+
+Stand-alone HTML files are often used in Jenkins for things like inline
+help messages. To translate these resources, add the
+locale code between the file name and the extension. For example, the
+Japanese version of `+abc.html+` would be `+abc_ja.html+`, and the British
+version of it could be `+abc_en_GB.html+`. These files must be
+encoded in UTF-8.
+
+=== Pushing changes
+
+After making some changes, you can commit them. Translators should
+consider themselves as owning property files for their locale, so feel
+free to go ahead and just commit. If you are new to this, doing a small
+commit first is a good idea. You can instead send in a patch if you
+prefer to be safe.
```

Comment:
This advice is from the days of subversion, before we used git and pull requests.  Let's update it to refer to pull requests and Crowdin where that applies.
```suggestion
After making changes, please test them to confirm they are working as expected.
After successfully testing the changes interactively, submit a pull request to propose the changes.
Instructions for localization pull requests to Jenkins core are available in the link:https://github.com/jenkinsci/jenkins/blob/master/CONTRIBUTING.md#proposing-changes[Jenkins core contributing guide].
Plugins that use Crowdin for translation can follow the link:/doc/developer/crowdin/translating-plugins/[Crowdin translation instructions].
Plugins that do not use Crowdin for translation should submit a pull request according to the contributing guide for the specific plugin.
```

Reply:
Fixed.